test/lib/tasks/generateThemeDesignerResources.js

```diff
@@ -0,0 +1,482 @@
+			}
+		})
+	});
+
+	const workspace = {
+		byGlob: sinon.stub().callsFake(async (globPattern) => {
+			if (globPattern === "/resources/**/themes/*/library.source.less") {
+				return [librarySourceLessResource1, librarySourceLessResource2, librarySourceLessResource3];
+			} else {
+				return [];
+			}
+		}),
+		write: sinon.stub()
+	};
+	const dependencies = {};
+
+	const libraryLessResource1 = {};
+	const libraryLessResource2 = {};
+	const libraryLessResource3 = {};
+
+	libraryLessGeneratorStub.resolves([libraryLessResource1, libraryLessResource2, libraryLessResource3]);
+
+	await generateThemeDesignerResources({
+		workspace,
+		dependencies,
+		options: {
+			projectName: "sap.ui.demo.lib",
+			version: "1.2.3",
+			namespace: "sap/ui/demo/lib"
+		}
+	});
+
+	t.is(t.context.ReaderCollectionPrioritizedStub.callCount, 1, "ReaderCollectionPrioritized should be created once");
+	t.deepEqual(t.context.ReaderCollectionPrioritizedStub.getCall(0).args, [{
+		name: `generateThemeDesignerResources - prioritize workspace over dependencies: sap.ui.demo.lib`,
+		readers: [workspace, dependencies]
+	}]);
+	const combo = t.context.ReaderCollectionPrioritizedStub.getCall(0).returnValue;
+
+	t.is(fsInterfaceStub.callCount, 1, "fsInterface should be created once");
+	t.deepEqual(fsInterfaceStub.getCall(0).args, [combo], "fsInterface should be created for 'combo'");
+	const fs = fsInterfaceStub.getCall(0).returnValue;
+
+	t.is(libraryLessGeneratorStub.callCount, 1);
+
+	t.deepEqual(libraryLessGeneratorStub.getCall(0).args[0], {
+		resources: [librarySourceLessResource1, librarySourceLessResource2, librarySourceLessResource3],
+		fs,
+	}, "libraryLessGenerator processor should be called with expected arguments");
+
+	t.is(ResourceStub.callCount, 3);
+	t.true(ResourceStub.alwaysCalledWithNew());
+
+	t.deepEqual(ResourceStub.getCall(0).args, [{
+		path: "/resources/sap/ui/demo/lib/.theming",
+		string: JSON.stringify({
+			sEntity: "Library",
+			sId: "sap/ui/demo/lib",
+			sVersion: "1.2.3"
+		}, null, 2)
+	}]);
+	const libraryDotTheming = ResourceStub.getCall(0).returnValue;
+
+	t.deepEqual(ResourceStub.getCall(1).args, [{
+		path: "/resources/sap/ui/demo/lib/themes/base/.theming",
+		string: JSON.stringify({
+			sEntity: "Theme",
+			sId: "base",
+			sVendor: "SAP"
+		}, null, 2)
+	}]);
+	const baseThemeDotTheming = ResourceStub.getCall(1).returnValue;
+
+	t.deepEqual(ResourceStub.getCall(2).args, [{
+		path: "/resources/sap/ui/demo/lib/themes/my_theme/.theming",
+		string: JSON.stringify({
+			sEntity: "Theme",
+			sId: "my_theme",
+			sVendor: "SAP",
+			oExtends: "base"
+		}, null, 2)
+	}]);
+	const myThemeDotTheming = ResourceStub.getCall(2).returnValue;
+
+	t.is(clonedCoreBaseDotThemingResourceStub.setPath.callCount, 1);
+	t.deepEqual(clonedCoreBaseDotThemingResourceStub.setPath.getCall(0).args,
+		["/resources/sap/ui/demo/lib/themes/sap_fiori_3/.theming"]);
+
+	t.is(workspace.write.callCount, 7);
+	t.is(workspace.write.getCall(0).args.length, 1);
```

Comment:
I would appreciate some assertion messages here. Especially in those cases where no named variable is used.